src/controllers/country.js

```diff
@@ -0,0 +1,157 @@
+      if (error) return res.status(400).json({ status: 400, error: error.message });
+      const country = await Admin.checkCountry(countryName);
+      if (country) return res.status(409).json({ status: 409, message: "This country already exists in the database." });
+      const createdCountry = await Admin.addCountry(newCountry);
+      return res.status(201).json({ status: 201, message: "A country has been added.", data: createdCountry, });
+    } catch (error) {
+      return res.status(500).json({ status: 500, error: "Server error." });
+    }
+  }
+
+  /**
+   * gets a list of all countries
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} list of countries
+   */
+  static async listCountries(req, res) {
+    try {
+      const countries = await db.Countries.findAll({
+        attributes: countriesAttributes,
+      });
+      res.status(200).send({
+        status: 200,
+        message: "Successfully retrived all countries",
+        data: countries,
+      });
+    } catch (error) {
+      res.status(500).send({
+        status: 500,
+        error,
+      });
+    }
+  }
+
+  /**
+   * gets a country with a specific id
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} country with specific id
+   */
+  static async getCountry(req, res) {
+    const { id } = req.params;
+    try {
+      const country = await getOneCountry(id);
+      return res.status(200).send({
+        status: 200,
+        message: `Successfully retrived country with id ${id}`,
+        data: country,
+      });
+    } catch (error) {
+      return res.status(404).send({
+        status: 404,
+        error: `Country with  id '${id}' not found`,
+      });
+    }
+  }
+
+  /**
+    * deletes a country with the specified id
+    * @param {object} req - request object
+    * @param {object} res - response object
+    * @returns {object} - result
+    */
+  static async deleteCountry(req, res) {
+    const { id } = req.params;
+    try {
+      const country = await getOneCountry(id);
+
+      // delete country from database
+      await country.destroy({ cascade: true });
+
+      return res.status(200).send({
+        status: 200,
+        message: `Successfully deleted country with id ${id}`,
+      });
+    } catch (error) {
+      return res.status(404).send({
+        status: 404,
+        error: `Country with  id '${id}' not found`,
+      });
+    }
+  }
+
+  /**
+    * updates a country with a specified id
+    * @param {object} req - request object
+    * @param {object} res - response object
+    * @returns {object} - result
+    */
+  static async updateCountry(req, res) {
+    const { id } = req.params;
+    try {
+      await db.Countries.update(req.body, {
+        where: { id },
+      });
+      return res.status(200).send({
+        status: 200,
+        message: `Successfully updated country with id ${id}`,
```

Comment:
in the messages, i do not like this `{
    "status": 200,
    "message": "Successfully deleted country with id ed5b34c0-f039-4526-ba7f-c8659001bd0b"
}`

just go with `{
    "status": 200,
    "message": "Successfully deleted country (no need to put the id)"
}`
 for all the success and error messages